### DISCUSSION TITLE
Two of my independent applications are now experiencing Task bar Pinning issues through their use of CefSharp

### DISCUSSION THREAD

**Discussion by llexel** [cefsharp/CefSharp#4972] [General]
### Is there an existing issue for this?

- [X] I have searched both open/closed issues, no issue already exists.

### CefSharp Version

127.3.50

### Operating System

Windows 10

### Architecture

x86

### .Net Version

.Net 4.8

### Implementation

WinForms

### Reproduction Steps

Upgrading from v125.0.210 to v127.3.50 caused many of our users to start reporting issues.   We have an outlook addin running cefsharp and a winforms application running cefsharp too, and suddenly these two pinned taskbar applications are combining into one and people can't find Microsoft Outlook because it's hidden under our other applications icon.   It's not even supposed to be possible to change an outlook icon from a plugin, but clearly CefSharp is managing it.

I even tried putting in settings.CefCommandLineArgs.Add("app-user-model-id", "Microsoft.Office.OUTLOOK.EXE"); before intialising the browser for the outlook app only but it made no difference.  I tried calling shell32 SetCurrentProcessExplicitAppUserModelID from both outlook and my other app, nothing I try works around the issue.  Somehow something in chromium or cefsharp has triggered this between these versions, and I'm not sure what else I can do

### Expected behavior

Cefsharp is not supposed to change an applications icon or its pin

### Actual behavior

Outlook is being unpinned and shown under a completely different application simply because an outlook COM addin I develop runs cefsharp and my application does too.

### Regression?

Worked fine in v125.0.210 (and before), still broken in v129.0.11

### Known Workarounds

Cannot figure out a workaround other than rolling back

### Does this problem also occur in the CEF Sample Application

Not Tested

### Other information

The log all works normally, there is no crash as such

**Comment by amaitland** [comment]
Converting to discussion as this is unlikely that `CefSharp` is doing directly.

> Upgrading from v125.0.210 to v127.3.50

There are **major** upstream changes that are most likely the cause in behavior change.

In `127` try changing back to the alloy bootstrap to see if the old behavior is restored.

https://github.com/cefsharp/CefSharp/issues/4835

Make sure to read https://groups.google.com/g/cef-announce/c/s1WaovAopFo/m/LV5eiNX1BgAJ

> Somehow something in chromium or cefsharp

It's almost certainly in `Chromium`. Are you using `Alloy` or `Chrome` styled browsers?

https://www.magpcss.org/ceforum/index.php is where I'd be asking for help on this.

> I even tried putting in settings.CefCommandLineArgs.Add("app-user-model-id", "Microsoft.Office.OUTLOOK.EXE");

Where did you get `app-user-model-id` from? I don't see any such command line arg when searching the `Chromium` source.

**Comment by johnjwalls** [comment]
We are having the same exact problem.  Two separate applications we have that both use CefSharp "merge" on the taskbar if left dormant for a few minutes.  When we revert to an older version (tagged as "vulnerable" by most lists), the behavior goes away.

**Comment by amaitland** [comment]
There was an upstream issue raised, I suggest you join in there. Looks like additional testing was requested that has yet to be completed.

https://github.com/chromiumembedded/cef/issues/3830

**Reply by amaitland** [reply]
Someone has created a new issue and has provided a workaround.

https://github.com/chromiumembedded/cef/issues/3977

As expected this will need to be fixed upstream. Follow the conversation there.
